docs/sophia.md

```diff
@@ -0,0 +1,1094 @@
+        - [Strings](#strings)
+        - [Byte arrays](#byte-arrays)
+        - [Cryptographic builins](#cryptographic-builins)
+            - [AEVM note](#aevm-note)
+        - [Authorization interface](#authorization-interface)
+        - [Oracle interface](#oracle-interface)
+            - [Example](#example)
+            - [Sanity checks](#sanity-checks)
+        - [AENS interface](#aens-interface)
+        - [Events](#events)
+            - [Argument order](#argument-order)
+        - [Compiler pragmas](#compiler-pragmas)
+        - [Exceptions](#exceptions)
+    - [Syntax](#syntax)
+        - [Lexical syntax](#lexical-syntax)
+            - [Comments](#comments)
+            - [Keywords](#keywords)
+            - [Tokens](#tokens)
+        - [Layout blocks](#layout-blocks)
+        - [Notation](#notation)
+        - [Declarations](#declarations)
+        - [Types](#types-1)
+        - [Statements](#statements)
+        - [Expressions](#expressions)
+        - [Operators types](#operators-types)
+        - [Operator precendences](#operator-precendences)
+    - [Examples](#examples)
+    - [The lifetime of a contract](#the-lifetime-of-a-contract)
+        - [Killing a contract](#killing-a-contract)
+
+## The Sophia Language
+An Æternity BlockChain Language
+
+The Sophia is a language in the ML family. It is strongly typed and has
+restricted mutable state.
+
+Sophia is customized for smart contracts, which can be published
+to a blockchain (the Æternity BlockChain). Thus some features of conventional
+languages, such as floating point arithmetic, are not present in Sophia, and
+some blockchain specific primitives, constructions and types have been added.
+## Language Features
+### Contracts
+
+The main unit of code in Sophia is the *contract*.
+
+- A contract implementation, or simply a contract, is the code for a
+  smart contract and consists of a list of types, entrypoints and local
+  functions. Only the entrypoints can be called from outside the contract.
+- A contract instance is an entity living on the block chain (or in a state
+  channel). Each instance has an address that can be used to call its
+  entrypoints, either from another contract or in a call transaction.
+- A contract may define a type state encapsulating its local
+  state. When creating a new contract the `init` entrypoint is executed and the
+  state is initialized to its return value.
+  
+The language offers some primive functions to interact with the blockchain and contracts.
+Please refer to the [Chain](sophia_stdlib.md#Chain), [Contract](sophia_stdlib.md#Contract)
+and the [Call](sophia_stdlib.md#Call) namespaces in the documentation.
+
+#### Calling other contracts
+
+To call a function in another contract you need the address to an instance of
+the contract. The type of the address must be a contract type, which consists
+of a number of type definitions and entrypoint declarations. For instance,
+
+```javascript
```

Comment:
Is `javascript` the best annotation? Shouldn't we add Sophia to github instead :-)

Reply:
Not my idea, but seems to work more or less :p

Reply:
@radrow Maybe you can take over and fix https://github.com/github/linguist/pull/4719 ? @thepiwo